### PR TITLE
Add instructions for pulling KPO images from GCP Artifact Registry

### PR DESCRIPTION
**Pull request by stuart23**
This is a feature that one customer has been using, I would like to have it documented

### CODE REVIEW

File: astro/kubernetespodoperator.md [hunk 1]
```diff
@@ -144,14 +147,23 @@ example_volume_test = KubernetesPodOperator(
  
 ## Run images from a private registry
 
-By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, you need to create a Kubernetes Secret and then specify the Kubernetes Secret in your DAG. If your Docker image is hosted in an Amazon Elastic Container Registry (ECR) repository, see [Docker images hosted in private Amazon ECR repositories](#docker-images-hosted-in-private-amazon-ecr-repositories).
+By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If your images are hosted on the container registry native to your cloud provider, you can grant access to the images directly. Otherwise, if you are using any other private registry, you need to create a Kubernetes Secret and then specify the Kubernetes Secret in your DAG.
+
+<Tabs
+    defaultValue="Private Registry"
+    groupId="registry-options"
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:154]:
```suggestion
    groupId="run-images-from-a-private-registry"
```

File: astro/kubernetespodoperator.md [hunk 2]
```diff
@@ -229,6 +246,42 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
     - In the Amazon ECR Dashboard, click **Repositories** in the left menu.
     - Click the **Private** tab and then copy the URI of the repository that hosts the Docker image.
 
+</TabItem>
+
+<TabItem value="Google Artifact Registry">
+
+### Docker images hosted in private Google Artifact Registry repositories
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:253]:
```suggestion
```
Is this redundant with the tab title?

File: astro/kubernetespodoperator.md [hunk 3]
```diff
@@ -185,6 +197,11 @@ KubernetesPodOperator(
     get_logs=True,
 )
 ```
+
+</TabItem>
+
+<TabItem value="AWS ECR">
+
 ### Docker images hosted in private Amazon ECR repositories
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:205]:
```suggestion
```

File: astro/kubernetespodoperator.md [hunk 4]
```diff
@@ -229,6 +246,42 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+
+This setup is available only on Astro Hybrid. 
+
+:::
+
+If your Docker image is hosted in Google Artifact Registry repository, add a permissions policy to the repository to allow the KubernetesPodOperator to pull the Docker image. You don't need to create a Kubernetes secret, or specify the Kubernetes secret in your DAG. Docker images hosted in Google Artifact Registry repositories can only be pulled from GCP clusters.
+
+### Prerequisites
+
+- Contact [Astronomer support](https://support.astronomer.io) to request the Compute Engine default service account ID for your cluster.
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:265]:
```suggestion
```

File: astro/kubernetespodoperator.md [hunk 5]
```diff
@@ -229,6 +246,42 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+
+### Steps
+
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:268]:
```suggestion

1. Contact [Astronomer support](https://support.astronomer.io) to request the Compute Engine default service account ID for your cluster.
```

File: astro/kubernetespodoperator.md [hunk 6]
```diff
@@ -229,6 +246,42 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+3. In the properties pane that appears, click **ADD PRINCIPAL** on the **PERMISSIONS** tab.
+4. In the Add Principals text box, copy in the Compute Engine default service account ID that was supplied by Astronomer Support
+5. In the Assign Roles selector, type "Artifact Registry Reader" and then select that item from the filtered list.
+6. Click **Save** to grant read access to Astro.
+7. [Set up the KubernetesPodOperator](#set-up-the-kubernetespodoperator).
+8. Replace `<your-docker-image>` in the instantiation of the KubernetesPodOperator with the Google Artifact Registry image URI. To locate the URI:
+
+    - In the Google Artifact Registry, click on the registry containing the image to open it.
+    - Click on the image you wish to use to open it.
+    - Click on the copy icon next to the image in the top corner. This will copy a string in the format `<GCP Region>-docker.pkg.dev/<Project Name>/<Registry Name>/<Image Name>`
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:280]:
```suggestion
2. Log in to Google Artifact Registry.
3. Click the checkbox next to the repository that you want to use.
4. In the **Properties** pane that appears, click **ADD PRINCIPAL** in the **PERMISSIONS** tab.
5. In the **Add Principals** text box, paste the Compute Engine default service account ID that was provided to you by Astronomer Support.
6. In the **Assign Roles** selector, search for `Artifact Registry Reader` and select the role that appears. 
7. Click **Save** to grant read access for the registry  to Astro.
8. [Set up the KubernetesPodOperator](#set-up-the-kubernetespodoperator). When you configure an instantiation of the KubernetesPodOperator, replace `<your-docker-image>` with the Google Artifact Registry image URI. To retrieve the URI:

    - In the Google Artifact Registry, click the registry containing the image.
    - Click the image you want to use.
    - Click the copy icon next to the image in the top corner. The string you copy should be in the format `<GCP Region>-docker.pkg.dev/<Project Name>/<Registry Name>/<Image Name>`.
```

File: astro/kubernetespodoperator.md [hunk 7]
```diff
@@ -229,6 +246,42 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+
+### Docker images hosted in private Google Artifact Registry repositories
+
+:::info
+
+This setup is available only on Astro Hybrid. 
+
+:::
+
+If your Docker image is hosted in Google Artifact Registry repository, add a permissions policy to the repository to allow the KubernetesPodOperator to pull the Docker image. You don't need to create a Kubernetes secret, or specify the Kubernetes secret in your DAG. Docker images hosted in Google Artifact Registry repositories can only be pulled from GCP clusters.
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:261]:
```suggestion
If your Docker image is hosted in Google Artifact Registry repository, add a permissions policy to the repository to allow the KubernetesPodOperator to pull the Docker image. You don't need to create a Kubernetes secret or specify the Kubernetes secret in your DAG. Docker images hosted in Google Artifact Registry repositories can be pulled only to Deployments hosted on GCP clusters.
```

**Review by jwitz — APPROVED** [review]
LGTM! Just some minor wording and format changes

**Comment by stuart23** [comment]
Thanks @jwitz for all the suggestions. Looks good to merge now.

**Pull request merged**